torch/nn/parallel/moe.py

```diff
@@ -16,7 +16,7 @@ def _get_occurrence_numbers(topk_indices: torch.Tensor) -> torch.Tensor:
     unique_vals, inverse = torch.unique(topk_indices, return_inverse=True)
 
     # Create a tensor to count occurrences for each unique value
-    n_unique = len(unique_vals)
+    n_unique = n_experts
     n_elements = len(topk_indices)
 
     # Create a matrix where each row corresponds to a unique value
```

Comment:
Why do we need torch.arange as an accessory below, why can't we just do [inverse, :] ?